ex5B/ex5.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * Write a function that given a string, returns the first character that
+ * doesn't repeat in the whole string. Otherwise (all characters repeat),
+ * should return null.
+ *
+ * For example:
+ * firstUniqueChar("abcdedcba") === "e"
+ * firstUniqueChar("abcdeedcba") === null
+ * firstUniqueChar("abababcab") === "c"
+ * firstUniqueChar("aaaaaaaaa") === null
+ * firstUniqueChar("a") === "a"
+ */
+function firstUniqueChar(str) {
+
+  var objCharCounter = {}, objFirstChar = { charVal: null }, searchIndex = 0;
+
+  for (searchIndex = 0; searchIndex < str.length; searchIndex++) {
+
+    if (objCharCounter[str.charAt(searchIndex)] === undefined) {
+
+      // Create new property
+      objCharCounter[str.charAt(searchIndex)] = { index: searchIndex, count: 1};
```

Comment:
While the property `index` is fine, it is not really accurate when you have `count >1`. I would rename it to `firstIndex`